secret_store/src/listener/service_contract.rs

```diff
@@ -0,0 +1,314 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::sync::{Arc, Weak};
+use futures::{future, Future};
+use parking_lot::RwLock;
+use ethcore::filter::Filter;
+use ethcore::client::{Client, BlockChainClient, BlockId};
+use ethkey::{Public, Signature, public_to_address};
+use ethsync::SyncProvider;
+use native_contracts::SecretStoreService;
+use hash::keccak;
+use bigint::hash::H256;
+use bigint::prelude::U256;
+use listener::service_contract_listener::ServiceTask;
+use {ServerKeyId, NodeKeyPair, ContractAddress};
+
+/// Name of the SecretStore contract in the registry.
+const SERVICE_CONTRACT_REGISTRY_NAME: &'static str = "secretstore_service";
+
+/// Key server has been added to the set.
+const SERVER_KEY_REQUESTED_EVENT_NAME: &'static [u8] = &*b"ServerKeyRequested(bytes32,uint256)";
```

Comment:
For future: it should be in generated code, might be good to migrate to `ethabi_derive` some time soon.